lib/open_food_network/orders_and_fulfillments_report.rb

```diff
@@ -238,31 +282,55 @@ def columns
           proc { |line_items| line_items.first.order.bill_address.andand.state }
         ]
       else
-        [proc { |line_items| line_items.first.product.supplier.name },
-         proc { |line_items| line_items.first.product.name },
-         proc { |line_items| line_items.first.full_name },
-         proc { |line_items| line_items.sum(&:quantity) },
-         proc { |line_items| line_items.first.price },
-         proc { |line_items| line_items.sum { |li| li.quantity * li.price } },
-         proc { |_line_items| "" },
-         proc { |_line_items| I18n.t(:report_header_incoming_transport) }]
+        DefaultReport.new(self).columns
       end
     end
 
     private
 
+    def supplier_name
+      proc { |line_items| find_variant(line_items.first.variant_id).product.supplier.name }
+    end
+
+    def product_name
+      proc { |line_items| find_variant(line_items.first.variant_id).product.name }
+    end
+
+    def line_item_name
+      proc { |line_item| find_variant(line_item.variant_id).full_name }
+    end
+
+    def line_items_name
+      proc { |line_items| find_variant(line_items.first.variant_id).full_name }
+    end
+
     def permissions
       return @permissions unless @permissions.nil?
       @permissions = OpenFoodNetwork::Permissions.new(@user)
     end
 
     def total_units(line_items)
-      return " " if line_items.map{ |li| li.unit_value.nil? }.any?
+      return " " if not_all_have_unit?(line_items)
+
       total_units = line_items.sum do |li|
-        scale_factor = ( li.product.variant_unit == 'weight' ? 1000 : 1 )
-        li.quantity * li.unit_value / scale_factor
+        product = find_variant(li.variant_id).product
+        li.quantity * li.unit_value / scale_factor(product)
```

Comment:
Should this be `scale_factor(li.product)`?

Reply:
And is this covered by tests or not?

Reply:
product refers to the local var above. I can't tell now if this is covered but it's very likely.

Reply:
Matt, here we cant refer directly to li.product because we want deleted products to be used, and that's why we need to go through find_variant.